lib/minstall.js

```diff
@@ -133,315 +118,586 @@ function checkStartConditions() {
-      logger.debug('linking local modules to root-node_modules');
-      return Promise.all(moduleInfos.map((moduleInfo) => {
+function setupLogger() {
+  logger.remove(logger.transports.Console);
+  logger.add(logger.transports.Console, {
+    stderrLevels: ['warn', 'error', 'critial'],
+    colorize: true,
+    handleExceptions: true,
+    humanReadableUnhandledException: true,
+    timestamp: false,
+    prettyPrint: true,
+  });
 
-        // local modules should be linked using the folder-names they should have,
-        // no matter what folder-name they actually have, therefore don't use .realFolderName here
-        const targetPath = path.join(cwd, 'node_modules', moduleInfo.folderName);
-        const modulePath = path.join(cwd, moduletools.modulesFolder, moduleInfo.realFolderName);
-        return systools.link(modulePath, targetPath);
-      }));
+  const logLevels = {
+    critical: {level: 0, color: 'red'},
+    error: {level: 1, color: 'magenta'},
+    warn: {level: 2, color: 'yellow'},
+    info: {level: 3, color: 'green'},
+    verbose: {level: 4, color: 'gray'},
+    debug: {level: 5, color: 'blue'},
+    silly: {level: 6, color: 'cyan'},
+  };
+  const levels = {};
+  const colors = {};
+
+  Object.keys(logLevels)
+    .forEach((name) => {
+      levels[name] = logLevels[name].level;
+      colors[name] = logLevels[name].color;
     });
-}
 
-function linkRootToModule(moduleFolder) {
-  logger.debug('linking root-node_modules to module-node_modules');
-  const rootModuleFolder = path.join(cwd, 'node_modules');
-  const targetFolder = path.join(cwd, moduletools.modulesFolder, moduleFolder, 'node_modules');
-  return systools.link(rootModuleFolder, targetFolder);
+  logger.setLevels(levels);
+  logger.addColors(colors);
 }
 
-function installMissingModuleDependencies(moduleFolder, missingDependencies) {
-  logger.debug('install missing dependencies');
-  const moduleNodeModules = path.join(cwd, moduletools.modulesFolder, moduleFolder, 'node_modules');
-  return systools.delete(moduleNodeModules)
-    .then(() => {
-      return moduletools.installPackets(missingDependencies);
-    });
-}
+function findRequestedDependencies(localModules) {
+  const requestedDependencies = {};
+  for (const module of localModules) {
+    for (const dependency in module.dependencies) {
+      const requestedVersion = module.dependencies[dependency];
+      if (requestedDependencies[dependency] === undefined) {
+        requestedDependencies[dependency] = {};
+      }
 
-function runPostinstall(moduleInfo) {
-  if (!moduleInfo.postinstall) {
-    logger.debug('won\'t run postinstall because there is no postinstall script');
-    return Promise.resolve();
-  }
+      let versionFound = false;
+      for (const version in requestedDependencies[dependency]) {
+        let intersection = null;
+        try {
+          intersection = intersect(requestedVersion, version);
+        } catch (error) {
+          // the versions didn't intersect. That's ok!
```

Comment:
it will raise an error if the versions do not intersect?

What is a version Intersection?

Reply:
probably call `error`: `noIntersectingVersionsFoundError`

Reply:
I'm using the module [semver-intersect](https://www.npmjs.com/package/semver-intersect). i give it two semver-ranges as input, and it returns a semver-range that satisfies both input-ranges. if the input-ranges don't intersect, the module throws an Error.

The fact that the versions don't intersect is not an error from minstalls point of view though, therefore nothing happens here, and i put the comment that tells you that it's ok for versions to not intersect